skbeam/core/roi.py

```diff
@@ -397,9 +397,8 @@ def mean_intensity(images, labeled_array, index=None):
         mean_intensity[n] = ndim.mean(img, labeled_array, index=index)
     return mean_intensity, index
 
-
-def circular_average(image, calibrated_center, threshold=0, nx=100,
-                     pixel_size=(1, 1),  min_x=None, max_x=None):
+def circular_average(image, calibrated_center, threshold=0, nx=None,
+                     pixel_size=(1, 1),  min_x=None, max_x=None, mask=None):
```

Comment:
@yugangzhang yes I agree with you `mask is required to make a correct normalization of circular average.`. But you can take care of that outside of this function. that means do this step before using the circular average
`image_mask =     np.array( image )[mask] `